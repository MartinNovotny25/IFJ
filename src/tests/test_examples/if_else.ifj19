a = 1

#IF <expr> : EOL INDENT <state> EOL <st-list> DEDENT   #mozna spatne, protoze chybi else
if a == 1:
    print('eq')
if a <= 1:
    print('less_eq')
if a >= 1:
    print('greater_eq')
if a != 1:
    print('not_eq')

#IF <expr> : EOL INDENT <state> EOL <st-list> DEDENT ELSE : EOL INDENT <state> EOL <st-list> DEDENT
if a == 1:
    print('eq')
else:
    print('else_eq')
if a <= 1:
    print('less_eq')
else:
    print('else_less_eq')
if a >= 1:
    print('greater_eq')
else:
    print('else_greater_eq')
if a != 1:
    print('not_eq')
else:
    print('else_not_eq')
