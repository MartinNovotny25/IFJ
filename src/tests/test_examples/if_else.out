TOKEN(IDENTIFIKATOR)
TOKEN(=)
TOKEN(INTEGER, 1)
TOKEN(EOL)
TOKEN(EOL)
TOKEN(EOL)
TOKEN(KLICOVE_SLOVO)
TOKEN(IDENTIFIKATOR)
TOKEN(==)
TOKEN(INTEGER, 1)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(IDENTIFIKATOR)
TOKEN(<=)
TOKEN(INTEGER, 1)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, less_eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(IDENTIFIKATOR)
TOKEN(>=)
TOKEN(INTEGER, 1)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, greater_eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(IDENTIFIKATOR)
TOKEN(!=)
TOKEN(INTEGER, 1)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, not_eq)
TOKEN())
TOKEN(EOL)
TOKEN(EOL)
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(IDENTIFIKATOR)
TOKEN(==)
TOKEN(INTEGER, 1)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, else_eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(IDENTIFIKATOR)
TOKEN(<=)
TOKEN(INTEGER, 1)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, less_eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, else_less_eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(IDENTIFIKATOR)
TOKEN(>=)
TOKEN(INTEGER, 1)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, greater_eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, else_greater_eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(IDENTIFIKATOR)
TOKEN(!=)
TOKEN(INTEGER, 1)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, not_eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(KLICOVE_SLOVO)
TOKEN(:)
TOKEN(EOL)
TOKEN(INDENT)
TOKEN(IDENTIFIKATOR)
TOKEN(()
TOKEN(STRING, else_not_eq)
TOKEN())
TOKEN(EOL)
TOKEN(DEDENT)
TOKEN(EOF)
